### DISCUSSION TITLE
How to create collection from custom model?

### DISCUSSION THREAD

**Discussion by sevenuz** [pocketbase/pocketbase#2008] [Q&A] [answered]
Hi, 
I created a custom model for my datatype as described in [docs](https://pocketbase.io/docs/custom-models/).

Now I want to create the collection from it in the init migration file. Therefore, I found
- https://github.com/pocketbase/pocketbase/blob/master/migrations/1640988000_init.go#L92
   - https://github.com/pocketbase/pocketbase/blob/master/models/collection.go

But there, the generic collection model is used. How do I init a collection with a custom model?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand what do you mean by _"init a collection with a custom model"_

You can't create a collection from a custom model struct, if that's what you are asking.
The custom model maps to an existing collection (or more precisely to its underlying records table).
All collections (no matter whether there is a custom record model or not) uses the same `models.Collection`.

To create manually/programmatically a collection you can check https://pocketbase.io/docs/collection-methods/#create-new-collection.

I'm not familiar with your use case and what you want to do, but if you enable the `Automigrate:true` during the [migrations initialization](https://pocketbase.io/docs/migrations/#enable-go-migrations), every collection change done from the Admin UI will generate automatically a Go migration for you.

If the above doesn't help, feel free to elaborate.
